optuna/samplers/_brute_force.py

```diff
@@ -130,10 +149,14 @@ def objective(trial):
             that it is not recommended using this option in distributed optimization settings since
             this option cannot ensure the order of trials and may increase the number of duplicate
             suggestions during distributed optimization.
+        enforce_unique_samples:
+            If this is :obj:`True`, the sampler strictly avoids duplicate samples. Please note that
+            the sampler may stop before trying the entire search space when this option is enabled.
     """
 
-    def __init__(self, seed: int | None = None) -> None:
+    def __init__(self, seed: int | None = None, enforce_unique_samples: bool = False) -> None:
```

Comment:
Setting `exclude_running = True` does not strictly avoid duplicate samples because we do not hold any lock on the storage (two workers may sample the same parameter at the same time and we have no way to strictly prevent that). So, a different name is preferred. How about "avoid_premature_stop"?